tests/Cluster/ClusterRepairsTest.cpp

```diff
@@ -0,0 +1,994 @@
+///
+/// @author Tobias Gödderz
+////////////////////////////////////////////////////////////////////////////////
+
+#include "catch.hpp"
+
+#include "Cluster/ClusterRepairs.h"
+#include "Cluster/ServerState.h"
+
+#include "Agency/AddFollower.h"
+#include "Agency/FailedLeader.h"
+#include "Agency/MoveShard.h"
+#include "lib/Random/RandomGenerator.h"
+
+#include <boost/core/demangle.hpp>
+#include <boost/date_time.hpp>
+#include <boost/range/combine.hpp>
+
+#include <iostream>
+#include <typeinfo>
+
+using namespace arangodb;
+using namespace arangodb::consensus;
+using namespace arangodb::cluster_repairs;
+
+namespace arangodb {
+
+bool operator==(AgencyWriteTransaction const& left,
+                AgencyWriteTransaction const& right) {
+  VPackBuilder leftBuilder;
+  VPackBuilder rightBuilder;
+
+  left.toVelocyPack(leftBuilder);
+  right.toVelocyPack(rightBuilder);
+
+  return velocypack::NormalizedCompare::equals(leftBuilder.slice(),
+                                               rightBuilder.slice());
+}
+
+std::ostream& operator<<(std::ostream& ostream,
+                         AgencyWriteTransaction const& trx) {
+  Options optPretty = VPackOptions::Defaults;
+  optPretty.prettyPrint = true;
+
+  VPackBuilder builder;
+
+  trx.toVelocyPack(builder);
+
+  ostream << builder.slice().toJson(&optPretty);
+
+  return ostream;
+}
+
+std::ostream& operator<<(std::ostream& ostream,
+                         std::list<RepairOperation> const& list) {
+  ostream << "std::list<RepairOperation> {\n";
+  if (!list.empty()) {
+    auto it = list.begin();
+    ostream << *it << "\n";
+    for (++it; it != list.end(); ++it) {
+      ostream << ", " << *it << "\n";
+    }
+  }
+  ostream << "}" << std::endl;
+  return ostream;
+}
+
+std::ostream& operator<<(std::ostream& ostream,
+                         std::vector<RepairOperation> const& vector) {
+  ostream << "std::vector<RepairOperation> {\n";
+  if (!vector.empty()) {
+    auto it = vector.begin();
+    ostream << *it << "\n";
+    for (++it; it != vector.end(); ++it) {
+      ostream << ", " << *it << "\n";
+    }
+  }
+  ostream << "}" << std::endl;
+  return ostream;
+}
+
+template <typename K, typename V>
+std::ostream& operator<<(std::ostream& ostream, std::map<K, V> const& map) {
+  std::string const typeNameK = boost::core::demangle(typeid(K).name());
+  std::string const typeNameV = boost::core::demangle(typeid(V).name());
+  ostream << "std::map<" << typeNameK << ", " << typeNameV << "> {\n";
+  if (!map.empty()) {
+    auto it = map.begin();
+    ostream << it->first << " => " << it->second << "\n";
+    for (++it; it != map.end(); ++it) {
+      ostream << ", " << it->first << " => " << it->second << "\n";
+    }
+  }
+  ostream << "}" << std::endl;
+  return ostream;
+}
+
+template <typename T>
+std::ostream& operator<<(std::ostream& stream, ResultT<T> const& result) {
+  std::string const typeName =
+      typeid(T) == typeid(RepairOperation)
```

Comment:
You like ternary operators don't you ;)

Reply:
I totally agree that this (as well as the other case) look awful. It looked way more readable in my original formatting. :disappointed: 
I'd like if/else and switch/case to be expressions, but as they're not, there is only the ternary op. ;)